pkg/apis/cdap/v1alpha1/cdapmaster_types.go

```diff
@@ -133,10 +133,21 @@ type UserInterfaceSpec struct {
 	CDAPExternalServiceSpec `json:",inline"`
 }
 
+// UpgradeJobSpec defines the specification for the upgrade job
+type UpgradeJobSpec struct {
+	Image    string `json:"image,omitempty"`
+	JobName  string `json:"jobName,omitempty"`
+	HostName string `json:"hostName,omitempty"`
+}
+
 // CDAPMasterStatus defines the observed state of CDAPMaster
 type CDAPMasterStatus struct {
-	status.Meta          `json:",inline"`
+	status.Meta `json:",inline"`
 	status.ComponentMeta `json:",inline"`
+	IsUpgrading                bool   `json:"isUpgrading"`
```

Comment:
Better use the `Condition` field in the `status.Meta` to report upgrade status. This is the common way in K8s. Upgrade failure can also be reported through that field.

BTW, when naming boolean field, name it without `is` prefix.